clevertap-core/src/test/java/com/clevertap/android/sdk/AnalyticsManagerTest.kt

```diff
@@ -1,96 +1,335 @@
+                context,
+                withArg {
+                    JSONAssert.assertEquals(json, it, true)
+                },
+                Constants.RAISED_EVENT
+            )
+        }
+
+        // setup again, 10000 ms has passed
+        every { timeProvider.invoke() } returns 12000
+
+        // Send duplicate PN
+        analyticsManagerSUT.pushNotificationClickedEvent(bundle)
+
+        // verify it was not called again, one time was from before
+        verify(exactly = 1) {
+            eventQueueManager.queueEvent(
+                context,
+                withArg {
+                    JSONAssert.assertEquals(json, it, true)
+                },
+                Constants.RAISED_EVENT
+            )
+        }
+        confirmVerified(eventQueueManager)
+    }
+
+    @Test
+    fun `clevertap processes PN clicked for same wzrk_id if separated by a span of greater than 5 seconds`() {
+
+        val json = notificationViewedJson(bundle);
+        every { timeProvider.invoke() } returns 10000
+
+        // send PN first time
+        analyticsManagerSUT.pushNotificationClickedEvent(bundle)
+
+        verify(exactly = 1) {
+            eventQueueManager.queueEvent(
+                context,
+                withArg {
+                    JSONAssert.assertEquals(json, it, true)
+                },
+                Constants.RAISED_EVENT
+            )
+        }
+
+        // setup again, 10000 ms has passed
+        every { timeProvider.invoke() } returns 20000
+
+        // Send duplicate PN
+        analyticsManagerSUT.pushNotificationClickedEvent(bundle)
+
+        // verify queue event called again
+        verify(exactly = 2) {
+            eventQueueManager.queueEvent(
+                context,
+                withArg {
+                    JSONAssert.assertEquals(json, it, true)
+                },
+                Constants.RAISED_EVENT
+            )
+        }
+        confirmVerified(eventQueueManager)
+    }
+
     @Test
     fun test_incrementValue_nullKey_noAction() {
         analyticsManagerSUT.incrementValue(null, 10)
 
-        verifyNoInteractions(validator)
+        verify {
+            validator wasNot called
+        }
     }
 
     @Test
     fun test_incrementValue_nullValue_noAction() {
         analyticsManagerSUT.incrementValue("abc", null)
 
-        verifyNoInteractions(validator)
+        verify {
+            validator wasNot called
+        }
     }
 
     @Test
     fun test_incrementValue_emptyKey_throwsEmptyKeyError() {
         mockCleanObjectKey("", 0)
         analyticsManagerSUT.incrementValue("", 10)
 
-        val captor = ArgumentCaptor.forClass(ValidationResult::class.java)
-        verify(validationResultStack).pushValidationResult(captor.capture())
-
-        assertEquals(512, captor.value.errorCode)
+        verify {
+            validationResultStack.pushValidationResult(withArg { arg ->
+                assertEquals(512, arg.errorCode)
```

Comment:
Do we have a constant for this and other error codes used in this test class?

Reply:
this is legacy test case, will fix when we look at unit testing task.